### PR TITLE
AN-888 Add Comparator Support in Select

### LINKED ISSUE #81: [FEATURE] Add support for comparison operator in SELECT clause
Add support for comparison operator in SELECT clause

### LINKED ISSUE #107: [BUG] Class not found exception when loading driver

### PR DESCRIPTION
**Pull request by xiazcy**
### Summary

AN-888 Add Comparator Support in Select

### Description

Support for single and multiple comparators (using AND/OR) inside the SELECT clause is added, i.e. `SELECT x = y FROM z`, `SELECT x = y OR x > 0 FROM z`.

### Related Issue

https://bitquill.atlassian.net/browse/AN-888
closes #81

### Additional Reviewers
@birschick-bq
@lyndonb-bq
@xiazcy
@simonz-bq
@alexey-temnikov
@kylepbit

### CODE REVIEW

File: sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/SqlMetadata.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 package org.twilmes.sql.gremlin.adapter.converter;
 
 import lombok.Getter;
+import lombok.Setter;
```
Comment (alexey-temnikov) [inline sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/SqlMetadata.java]: Is it used?

File: sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/GremlinSqlBasicCall.java [hunk 2]
```diff
@@ -83,6 +84,8 @@ public String getRename() throws SQLException {
                 // returns the formatted column name for aggregations
                 return ((GremlinSqlAggFunction) gremlinSqlOperator).getNewName();
             }
+        } else if (gremlinSqlOperator instanceof GremlinSqlBinaryOperator) {
+            return ((GremlinSqlBinaryOperator) gremlinSqlOperator).getNewName();
```
Comment (alexey-temnikov) [inline sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/GremlinSqlBasicCall.java:87]: Shall we also add AS support, e.g. - `SELECT age, (age < 18) as minor FROM person`?

File: sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/logic/GremlinSqlBinaryOperator.java [hunk 3]
```diff
@@ -176,6 +198,23 @@ void handleEmbeddedGremlinSqlBasicCall(GremlinSqlBasicCall gremlinSqlBasicCall,
...
+    public String getOperandName(final GremlinSqlNode operand) throws SQLException {
+        if (operand instanceof GremlinSqlIdentifier) {
+            final GremlinSqlIdentifier gremlinSqlIdentifier = (GremlinSqlIdentifier) operand;
+            return gremlinSqlIdentifier.isStar() ? "*" : gremlinSqlIdentifier.getColumn();
+        } else if (operand instanceof GremlinSqlLiteral) {
+            return ((GremlinSqlLiteral) operand).getValue().toString();
+        } else if (operand instanceof GremlinSqlBasicCall) {
+            return ((GremlinSqlBasicCall) operand).getRename();
+        }
+        throw new SQLException("Error, unsupported operand type, cannot rename column.");
```
Comment (alexey-temnikov) [inline sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/logic/GremlinSqlBinaryOperator.java]: Should we refine error, including what specific type is not supported? Would be good to include guidance on whether it is an error or not supported feature.

Comment (lyndonbauto) [reply sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/logic/GremlinSqlBinaryOperator.java]: Also add to bundle.

File: sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/SqlMetadata.java (hunk 1 above)
Comment (lyndonbauto) [reply sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/SqlMetadata.java]: Will remove

File: sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/GremlinSqlBasicCall.java (hunk 2 above)
Comment (lyndonbauto) [reply sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/GremlinSqlBasicCall.java:87]: Fixed.

Comment (lyndonbauto) [reply sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/operator/GremlinSqlBasicCall.java:87]: Fixed.

**Review by lyndonbauto — APPROVED** [review]

**Review by simonz-bq — APPROVED** [review]

**Pull request merged**